plugins/traefik/main.go

```diff
@@ -47,6 +62,13 @@ func (sm *SablierMiddleware) ServeHTTP(rw http.ResponseWriter, req *http.Request
 
 	conditonalResponseWriter := newResponseWriter(rw)
 
+	if isWebsocketRequest(req) {
+		// FIXME dynamic make no sense for websocket since client return error
+		fmt.Println("=== websocket request")
+		go monitorWebSocketActivity(sablierRequest, sm)
```

Comment:
I was thinking about this approach also. You create go routine per connection. I wondered if this would scale up. I didn't check details though.

Reply:
it should scalre correctly , maybe in next iteration , we should add Mutex for security , but not really conformatable with it

Reply:
I wrote a proxy for Sablier to handle TCP only services and had a similar problem: How to prevent the service from shutting down, if there are only long running connections.

I don't check if there is actual activity on the connection, but I have a single task/process that will, as long as there is at least 1 active connection, ping the Sablier endpoint in regular intervals. Maybe something like this would work here, too:

https://github.com/vbrandl/sablier-proxy/blob/main/src/main.rs#L154

This way, you don't have to spawn another goroutine per connection, but there will be exactly 1 monitoring task/routine, which might scale better if there are many connections.